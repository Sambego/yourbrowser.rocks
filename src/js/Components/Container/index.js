import {h, render, Component} from 'preact';
import {element} from 'proptypes';
import styles from './container.scss';

const Container = ({children}) => {
    return (
        <div className={styles.container}>
            {children}
        </div>
    );
};

Container.propTypes = {
    children: element.isRequired,
};

export default Container;
