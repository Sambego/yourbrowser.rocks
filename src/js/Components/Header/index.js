import {h, render, Component} from 'preact';
import styles from './header.scss';

const HEADER = () => {
    return (
        <header className={styles.header}>
            <h1 className={styles.header__title} >Your browser rocks!</h1>
        </header>
    );
};

HEADER.propTypes = {};

export default HEADER;
