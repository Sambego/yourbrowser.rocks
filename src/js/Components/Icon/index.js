import {h, render, Component} from 'preact';
import {bool, string} from 'proptypes';
import classnames from 'classnames';

const Icon = ({icon, color, flipHorizontal}) => {
    const classes = classnames(`iconic iconic-sm iconic-${icon} ${color}`);

    return <i className={classes} />;
};

Icon.propTypes = {
    color: string,
    icon: string.isRequired,
};

Icon.defaultProps = {
    color: '',
};

export default Icon;
