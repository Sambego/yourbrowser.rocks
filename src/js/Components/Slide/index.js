import {h, render, Component} from 'preact';
import {element, string} from 'proptypes';
import {SlideNav} from '../';
import styles from './slide.scss';

const Slide = ({children, previous, next}) => {
    return (
        <section className={styles.slide}>
            {previous && <SlideNav direction="left" to={previous} />}
            {next && <SlideNav direction="right" to={next} />}

            {children}
        </section>
    );
};

Slide.propTypes = {
    children: element.isRequired,
    previous: string,
    next: string,
};

export default Slide;
