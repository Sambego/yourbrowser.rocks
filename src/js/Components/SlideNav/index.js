import {h, render, Component} from 'preact';
import {element, string} from 'proptypes';
import classnames from 'classnames';
import {Icon} from '../';
import styles from './slide-nav.scss';

const SlideNav = ({direction, to}) => {
    const classes = classnames(styles['slide-nav'], {
        [styles['slide-nav--left']]: direction === 'left',
        [styles['slide-nav--right']]: direction === 'right',
    });

    return (
        <a className={classes} href={to}>
            <Icon icon={direction} />
        </a>
    );
};

SlideNav.propTypes = {
    direction: string,
    to: string,
};

export default SlideNav;
