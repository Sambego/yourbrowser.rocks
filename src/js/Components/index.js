export Container from './Container';
export Header from './Header';
export Slide from './Slide';
export SlideNav from './SlideNav';
export Icon from './Icon';
export Button from './Button';
