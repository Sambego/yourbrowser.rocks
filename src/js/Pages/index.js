export BatteryPage from './Battery';
export LocationPage from './Location';
