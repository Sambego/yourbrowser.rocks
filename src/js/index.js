import '../scss/styles.scss';

import {h, render} from 'preact';

render((
    <div>
        <h1>Your browser rocks!</h1>
    </div>
), document.body);
