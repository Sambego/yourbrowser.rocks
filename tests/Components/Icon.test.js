import render from 'preact-render-to-string';
import {h, render as PreactRender} from 'preact';
import {Icon} from '../../src/js/Components';

describe('Icon', () => {
    const icon = 'home';
    const color = 'tomato';

    it('should render correctly', () => {
        const iconElement = render(
            <Icon icon={icon} color={color}/>
        );

        expect(iconElement).toMatchSnapshot();
    });

    it('should add an icon class', () => {
        const iconElement = PreactRender(
            <Icon icon={icon}/>
        );

        expect(iconElement.classList.contains(`iconic-${icon}`)).toMatchSnapshot();
    });

    it('should add a color class', () => {
        const iconElement = PreactRender(
            <Icon icon={icon} color={color}/>
        );

        expect(iconElement.classList.contains(color)).toMatchSnapshot();
    });
});
