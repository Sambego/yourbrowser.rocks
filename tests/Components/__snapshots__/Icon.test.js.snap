exports[`Icon should add a color class 1`] = `true`;

exports[`Icon should add an icon class 1`] = `true`;

exports[`Icon should render correctly 1`] = `"<i class=\"iconic iconic-sm iconic-home tomato\"></i>"`;
